### PR TITLE
Improve classic optimiser

### PR DESCRIPTION
**Pull request by royerloic**
Lots of improvements, classic algos should optimise in a more robust way.

### CODE REVIEW

**Comment by royerloic** [comment]
Ready for your review

File: aydin/util/j_invariance/demo/demo_j_invariant_only_cat.py [hunk 1]
```diff
@@ -7,14 +7,12 @@
...
-from aydin.it.classic_denoisers.spectral import denoise_spectral
-from aydin.util.j_invariance.j_invariant_classic import calibrate_denoiser_classic
-from aydin.util.j_invariance.j_invariant_smart import calibrate_denoiser_smart
 from aydin.util.crop.rep_crop import representative_crop
+from aydin.util.j_invariance.j_invariance import calibrate_denoiser
 from aydin.util.log.log import Log
 
 
-def demo_j_invariant_classic(image, display=True):
+def demo_j_invariant_only_cat(image, display=True):
```
[the remark this answers is not in the document]
Comment (royerloic) [reply aydin/util/j_invariance/demo/demo_j_invariant_only_cat.py:15]: category

[the remark this answers is not in the document]
Comment (royerloic) [reply aydin/util/j_invariance/demo/demo_j_invariant_only_cat.py:15]: tests optimization over categories only

[the remark this answers is not in the document]
Comment (AhmetCanSolak) [reply aydin/util/j_invariance/demo/demo_j_invariant_only_cat.py:15]: thank you for clarification

**Comment by AhmetCanSolak** [comment]
I've started reviewing this PR Wednesday but some tests are failing, I will address those issues and get this PR merged today before the new version release.

**Pull request merged**